packages/client/hmi-client/src/components/responsive-matrix/cell-line-graph.vue

```diff
@@ -0,0 +1,261 @@
+			default() {
+				return [];
+			}
+		},
+		parameters: {
+			type: Array as PropType<string[]>,
+			default() {
+				return [];
+			}
+		},
+		parametersMin: {
+			type: Object,
+			default() {
+				return {};
+			}
+		},
+		parametersMax: {
+			type: Object,
+			default() {
+				return {};
+			}
+		},
+		colorFn: {
+			type: Function,
+			default() {
+				return '#000000';
+			}
+		}
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// data                                                                         //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	data() {
+		return {
+			topMargin: 10,
+			bottomMargin: 20,
+			leftMargin: 30,
+			rightMargin: 20,
+			containerBoundingBox: {} as DOMRect,
+			svg: null as any
+		};
+	},
+
+	// ---------------------------------------------------------------------------- //
+	// computed                                                                     //
+	// ---------------------------------------------------------------------------- //
+
+	// ///////////////////////////////////////////////////////////////////////////////
+
+	computed: {
+		selectedCells(): any {
+			const startRow = this.selectedCell[SelectedCellValue.START_ROW];
+			const endRow = this.selectedCell[SelectedCellValue.END_ROW];
+			const startCol = this.selectedCell[SelectedCellValue.START_COL];
+			const endCol = this.selectedCell[SelectedCellValue.END_COL];
+
+			// const selectedCells: any[] = [];
+			const selectedCells = this.parameters.reduce((acc, param) => {
+				acc[param] = [];
+				return acc;
+			}, {});
+
+			for (let row = startRow, i = 0; row <= endRow; row++, i++) {
+				this.dataRowList[row].forEach((cell: any) => {
+					if (cell.col <= endCol && cell.col >= startCol) {
+						this.parameters.forEach((param) => selectedCells[param].push(cell[param]));
+					}
+				});
+			}
+
+			return selectedCells;
+		},
+
+		labelColSelected(): number[] {
+			const startCol = this.selectedCell[SelectedCellValue.START_COL];
+			const endCol = this.selectedCell[SelectedCellValue.END_COL];
+			return this.labelColList.slice(startCol, endCol + 1);
+		},
+
+		// assume that timestep for all cells in same column are equal
+		timeMin(): number {
+			return Math.min(...this.labelColSelected);
+		},
+
+		// assume that timestep for all cells in same column are equal
+		timeMax(): number {
+			return Math.max(...this.labelColSelected);
+		},
+
+		parametersMinAll(): number {
+			return Math.min(...(Object.values(this.parametersMin) as number[]));
+		},
+
+		parametersMaxAll(): number {
+			return Math.max(...(Object.values(this.parametersMax) as number[]));
+		},
+
+		xScale(): any {
+			return scaleLinear()
+				.domain([this.timeMin, this.timeMax])
+				.range([0, this.containerBoundingBox.width - this.leftMargin - this.rightMargin]);
```

Comment:
If you are assuming that we are dealing with numbers here, you can get rid of `timeMax` and `timeMin` by just doing

```
return scaleLinear()
  .domain(d3.extent( ...this. labelColSelected))
  ...
```

e.g.
```
> d3.extent([1, 2, -9, 43, 99, 2, 1])
< (2) [-9, 99]
```

Reply:
ya this will work for now, while i hesitate to say we are only going to be dealing with numbers here i have not validated this thing for anything but numbers. in fact im pretty sure it would need lots of work to support string/categories as well

Reply:
ok i looked at this again, and i found two issues with removing timeMax and timeMin. one, is that timeMin/Max is computed over the entire visualization, so over all rows and columns. whereas using d3.extent will be finding the domain over only the selected dataset. this is important to make the scales of all graphs regardless of selection the same, so they can be compared visually if there are multiple drilldowns open.

the second (much lesser) reason is performance, calculating timeMin/Max comes for free with data ingestion (you are going to iterate through all the data anyway), using d3 extent adds some extra processing to opening the drilldown component.

what do you think?